File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int max=0;
        for(int i=0;i<s.length();i++)
        {
            String ch="";
            for(int j=i;j<s.length();j++)
            {
                char c=s.charAt(j);
                if(ch.contains(String.valueOf(c)))
                break;
                else
                ch+=c;
            }
            max=Math.max(max,ch.length());
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans=0
        l=0
        s1=set()
        for i in range(len(s)):
            while(s[i] in s1):
                s1.remove(s[l])
                l=l+1
            s1.add(s[i])
            ans=max(ans,len(s1))
        return ans

                
